pkgs/by-name/st/steel/package.nix

```
@@ -95,21 +95,27 @@ rustPlatform.buildRustPackage {
     STEEL_HOME = "${placeholder "out"}/lib/steel";
   };
 
-  passthru.updateScript = unstableGitUpdater {
-    tagConverter = writeShellScript "steel-tagConverter.sh" ''
-      export PATH="${
-        lib.makeBinPath [
-          curl
-          yq
-        ]
-      }:$PATH"
-
-      version=$(curl -s https://raw.githubusercontent.com/mattwparas/steel/refs/heads/master/Cargo.toml | tomlq -r .workspace.package.version)
-
-      read -r tag
-      test "$tag" = "0" && tag="$version"; echo "$tag"
-    '';
-  };
+  passthru.updateScript = _experimental-update-script-combinators.sequence [
+    (unstableGitUpdater {
+      tagConverter = writeShellScript "steel-tagConverter.sh" ''
+        export PATH="${
+          lib.makeBinPath [
+            curl
+            yq
+          ]
+        }:$PATH"
+
+        version=$(curl -s https://raw.githubusercontent.com/mattwparas/steel/refs/heads/master/Cargo.toml | tomlq -r .workspace.package.version)
+
+        read -r tag
+        test "$tag" = "0" && tag="$version"; echo "$tag"
+      '';
+    })
+    (nix-update-script {
+      # Updating `cargoHash`
+      extraArgs = [ "--version=skip" ];
+    })
+  ];
```

Wouldn't the following work just as well?

```suggestion
  passthru.updateScript = nix-update-script {
    extraArgs = [ "--version=branch" ];
  };
```

---

This would make the version `0-unstable-2025-06-14` instead of `0.6.0-unstable-2025-06-14`

---

Which is fine, as `steel` has no git tags. See https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning . There are tagged releases at [crates.io](https://crates.io/crates/steel-interpreter) but this only has `0.5.0`. I don't really understand why parsing `Cargo.toml` is needed in order to set _some_ version, which isn't really helpful anyways.